vignettes/wrappers.Rmd

```diff
@@ -36,45 +36,71 @@ This is the key move:
 
 If you'd like to understand why this is the way, keep reading.
 
-## A first attempt that does not work
+## Working example
 
-Here's a simple attempt at writing a wrapper around `glue()`:
+Here's an abbreviated excerpt of the roxygen comment that generates the documentation for the starwars dataset in dplyr:
+
+```r
+#' \describe{
+#' \item{name}{Name of the character}
+#' \item{height}{Height (cm)}
+#' \item{mass}{Weight (kg)}
+#' \item{species}{Name of species}
+#' \item{films}{List of films the character appeared in}
+#' }
+```
+
+To produce such text programmatically, the first step might be to generate the `\item` lines from a named list of column names and descriptions.
+Notice that `{` and `}` are important to the `\describe{...}` syntax, so this is an example where it is nice for glue to use different delimiters for expressions.
+
+Put the metadata in a suitable list:
 
 ```{r}
-myglue0 <- function(...) {
-  glue(..., .open = "@@", .close = "~~")
-}
+sw_meta <- list(
+  name    = "Name of the character",
+  height  = "Height (cm)",
+  mass    = "Weight (kg)",
+  species = "Name of species",
+  films   = "List of films the character appeared in"
+)
 ```
 
-From superficial experimentation, `myglue0()` appears to work:
+Define a custom glue wrapper and use it inside another helper that generates `\item` entries[^1]:
+
+[^1]: We're using `<@` and `@>` as the delimiters because this vignette is authored using R Markdown and knitr. The delimiters `<<` and ``>>` actually have special meaning in knitr, so it was easiest to use something else.
 
 ```{r}
-fn_def <- "
-  @@NAME~~ <- function(x) {
-    @@BODY~~
-  }"
+my_glue = function(...) {
+  glue(..., .open = "<@", .close = "@>", .envir = parent.frame())
+}
 
-myglue0(fn_def, NAME = "one_plus_one", BODY = "1 + 1")
+named_list_to_items <- function(x) {
+  my_glue("\\item{<@names(x)@>}{<@x@>}")
```

Comment:
Now that I think of it, is there a reason to not do just `<` and `>`?

Reply:
That would work in this exact example, but I don't want to model that in a vignette. I know some folks will copy exactly what they see here and I think `<` and `>` are too likely to collide with text in the template that should not get evaluated. Generally speaking.

Reply:
What about `((` or `[[`? I'm just thinking that `<@` looks hard to type.

Reply:
Well all of those come up a lot in real R code (like, the code you might want to evaluate) so again I shied away from them. It really sucks that you can't use `<<`. You can't even triple them `<<<` because that also matches knitr's regex (unintentionally, I think).